src/main/java/io/harness/cf/client/api/InnerClient.java

```diff
@@ -94,14 +88,60 @@ protected void setUp(@NonNull final Connector connector, @NonNull final BaseConf
     // initialization
     repository = new StorageRepository(options.getCache(), options.getStore(), this);
     evaluator = new Evaluator(repository);
-    authService = new AuthService(this.connector, options.getPollIntervalInSeconds(), this);
     pollProcessor =
         new PollingProcessor(this.connector, repository, options.getPollIntervalInSeconds(), this);
     metricsProcessor = new MetricsProcessor(this.connector, this.options, this);
     updateProcessor = new UpdateProcessor(this.connector, this.repository, this);
 
     // start with authentication
-    authService.startAsync();
+    authenticateAsync();
+  }
+
+  private void authenticateAsync() {
+    CompletableFuture.supplyAsync(this::authenticate);
+  }
+
+  private boolean authenticate() {
+
+    final int AUTH_TIMEOUT_MS = 60_000;
+    final long expireTime = System.currentTimeMillis() + AUTH_TIMEOUT_MS;
+    final int delayMs = ThreadLocalRandom.current().nextInt(5000, 10000);
+    boolean authenticated = false;
+    int attempt = 1;
+
+    do {
```

Comment:
It might be useful for anyone coming to this later to just add a comment in explaining the logic, i.e. the SDK attempts to authenticate, with a backing of delay, retrying up to 1 minute (set by expireTime)